lib/active_support/deprecation_test_helper.rb

```diff
@@ -1,10 +1,82 @@
 # frozen_string_literal: true
 
 require "active_support/deprecation_test_helper/version"
+require 'active_support/deprecation'
 
 module ActiveSupport
   module DeprecationTestHelper
-    class Error < StandardError; end
-    # Your code goes here...
+    class << self
+      def configure
+        defined?(Minitest) || defined?(RSpec) or raise 'Unexpected test suite encountered. This only supports Rspec and Minitest'
+
+        ActiveSupport::Deprecation.include(self)
+
+        if defined?(Minitest)
+          Minitest.after_run(after_all_callback)
+        elsif defined?(RSpec)
+          RSpec.configuration.after(:all, after_all_callback)
+        end
```

Comment:
I was suggesting an `else` here to raise an exception that we can't figure out the framework. If that happens, it's probably because they didn't require the framework before running this.

BTW, maybe pick off RSpec first? If they're both defined, I'd prefer that one. There might be cases where some legacy code pulled in MiniTest accidentally.

Also, no need to support Test::Unit?

Reply:
There is an early raise in the method to account for unsupported test setup rather than falling through the `if` statement.  This keeps things dry and removes any side affects of including the module in `ActiveSupport::Deprecation` and not having a reporting mechanism at the end.

I couldn't find a way to `mock` or `stub` the `defined?` method, so the only way to test to make sure that `RSpec` and `Minitest` are set up properly (since this tests using RSpec) is to put `Minitest` first.  With your comment though, we pull `RSpec::Mocks` into `Minitest` all the time.  So reordering would cause issues as well.  To fix this, I'm going to have this take an argument for what test environment to configure against.

In the majority of places I haven't seen much usage of `Test::Unit` so I left it out to keep initial scope low. It'll be easy to add later if we need it.

Reply:
Oh, I see the early raise now. That works, but it's obviously not DRY. In the event that someone tries this with only Test::Unit, do you really care whether you did the `ActiveSupport::Deprecation.include(self)` first? I don't think that would matter...in which case you can DRY it up by moving the assertion into an `else`.

Reply:
It does matter because it would have an unexpected side effect of partial configuration where all deprecation warnings are suppressed.  This is all moot though with the second paragraph in my response since this has all be refactored around taking an argument that specifies the test framework to configure against.  It's too common that test frameworks use some RSpec and some Minitest, so we need to know which is actually running the tests, which cannot be inferred.

Reply:
this is all done